script.js

```diff
@@ -1,4 +1,127 @@
+  console.log(rollDice, "dice roll");
+  return diceNumber;
+};
+
+var rollDiceForPlayer = function () {
+  console.log("Control flow : start of rollDiceForPlayer()");
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter = counter + 1;
+  }
+  console.log("rollDiceForPlayer Changes, playerRolls: ", currentPlayerRolls);
+  return `Welcome Player ${currentPlayer}<br>You rolled ${currentPlayerRolls[0]} for Dice 1 and ${currentPlayerRolls[1]} for Dice 2.<br>Choose the order of the dice.`;
+};
+
+var getPlayerScore = function (playerInput) {
+  var playerScore;
+  // input validation
+  if (playerInput != 1 && playerInput != 2) {
+    console.log(
+      "Control flow : input validation, invalid out is not 1 and not 2"
+    );
+    return `Error Please only onput '1' or '' to choose which dice to use as the first digit.<br> Your dice rolls are :<br>Dice 1 : ${currentPlayerRolls[0]} Dice 2 : ${currentPlayerRolls[1]}.`;
+  }
+  // input == 1
+  if (playerInput == 1) {
+    console.log("Control flow : input == 1");
+    var playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+    return `Your chosen value is: ${playerScore}.`;
+  }
+  // input == 2
+  if (playerInput == 2) {
+    console.log("Control flow : input == 2");
+    var playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+    // Store playerScore rolls array
+    allPlayerScore.push(playerScore);
+    // clear current player rolls array
+    currentPlayerRolls = [];
+    return `Player ${currentPlayer}, your chosen value is ${playerScore}`;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log("Checking game state on submit click: ".gameState);
+  console.log("Checking currentPlayer on submit click: ", currentPlayer);
+  var outputMessage = "";
+  if (gameState == GAME_STATE_DICE_ROLL) {
+    console.log("Control flow : gameState == GAME_STATE_DICE_ROLL");
+    // Display dice rolled as output message
+    outputMessage = rollDiceForPlayer();
+
+    // change game state
+    gameState = GAME_STATE_CHOOSE_DICE_ORDER;
+    return outputMessage;
+  }
+
+  if (gameState == GAME_STATE_CHOOSE_DICE_ORDER) {
+    console.log("Control flow : gameState == GAME_STATE_CHOOSE_DICE_ORDER");
+
+    // Call playerScore function
+    outputMessage = getPlayerScore(input);
+    if (currentPlayer == 1) {
+      console.log("Control flow : end of player 1, now player");
+      currentPlayer == 2;
+      gameState = GAME_STATE_DICE_ROLL;
+      //return outputMessage`<br>It is now player 2's turn!`;
+    }
+    if (currentPlayer == 2) {
+      console.log(
+        "Control flow : end of player 2s turn, Next submit click of calculate the score"
+      );
+      gameState = GAME_STATE_COMPARE_SCORES;
+      return outputMessage`<br> Press submit to caluclate scores`;
+    }
+  }
```

Comment:
a slight error here in line 114 (should be using = instead of ==) which causes your game to never update `currentPlayer`, resulting in `currentPlayer` to always be 1.

Reply:
i see you haven't had the chance to work on the logic for the subsequent game state `GAME_STATE_COMPARE_SCORES` so i cant test the other portions of your game even after fixing this bug